packages/cozy-device-helper/src/deeplink.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * This file is used to open the native app from a webapp
+ * if this native app is installed
+ *
+ * From a webapp, we don't have any clue if a native app is installed.
+ * The only way to know that, is to try to open the custom link
+ * (aka cozydrive://) and if nothing happens (no blur) we redirect to
+ * the callback
+ *
+ * Firefox tries to open custom link, so we need to create an iframe
+ * to detect if this is supported or not
+ */
+const _createHiddenIframe = (target, uri, randomId) => {
+  const iframe = document.createElement('iframe')
+  iframe.src = uri
+  iframe.id = `hiddenIframe_${randomId}`
+  iframe.style.display = 'none'
+  target.appendChild(iframe)
+  return iframe
+}
+
+const openUriWithHiddenFrame = (uri, failCb) => {
+  const randomId = Math.random()
+    .toString(36)
+    .replace(/[^a-z]+/g, '')
+    .substr(0, 5)
+  window.addEventListener('blur', onBlur)
+  const timeout = setTimeout(function() {
+    failCb()
+    window.removeEventListener('blur', onBlur)
+  }, 500)
+
+  let iframe = document.getElementById(`hiddenIframe_${randomId}`)
+  if (!iframe) {
+    iframe = _createHiddenIframe(document.body, 'about:blank', randomId)
+  }
+
+  function onBlur() {
+    clearTimeout(timeout)
+    window.removeEventListener('blur', onBlur)
+    iframe.parentElement.removeChild(iframe)
```

Comment:
We remove the iframe here, but we don't do it in the `timeout`, is that on purpose?